### ISSUE TITLE
proofs generated by zerokit can not be verified if the state of the Merkle tree of the sender (prover) differs from the receiver(verifier)

### ISSUE THREAD

**Issue by staheri14** [status-im/nwaku#1248]
### Problem
Scenario 1:
I am running  a waku node locally that listens to contract X, and a chat user that listens to another contract Y
Proofs generated by the chat user is considered invalid by the wakunode because of invalid proof
I do all of these on the latest master where root validation is removed

Scenario 2:
However, When I run the same scenario but with a difference, and that is both the waku node and the chat2 user listen to the same contract, the proof generated by chat2 gets verified by the waku node

My interpretation: the `proofVerify` is probably relying on the merkle tree internally (at the zerokit level) and verifies the proofs of messages against that, and that is why the first scenario (1) fails, because the waku node and the chat client do not have the same state of the Merkle tree.

~~### Impact~~
~~Indicate how significant you believe the impact of the bug is. Bugs that lead to data loss or corruption would be considered `critical`. In such cases, please also add the `critical` label.~~

### To reproduce
You can replicate the first scenario by this command:
run the wakunode as (use your own client address)
```
./build/wakunode2 --rln-relay:true  --rln-relay-dynamic:true  --staticnode:/ip4/127.0.0.1/tcp/60010/p2p/16Uiu2HAmKdCdP89q6CwLc6PeFDJnVR1EmM7fTgtphHiacSNBnuAz  --dns-discovery=false  --rln-relay-content-topic:/toy-chat/2/luzhou/proto --ports-shift=1 --nodekey=f157b19b13e9ee818acfc9d3d7eec6b81f70c0a978dec19def261172acbe26e6  --rln-relay-eth-contract-address:0x367F3e869cF2E1DD28644E308b42652cb6fcDA72  --rln-relay-eth-account-address:0000000000000000000000000000000000000000 --rln-relay-eth-client-address:xxx 
```
run the chat client as (please replace xxx values with your own inputs)
```
 ./build/chat2 --fleet=none --content-topic=/toy-chat/2/luzhou/proto --rln-relay=true --rln-relay-dynamic=true --rln-relay-eth-contract-address=0x4252105670fe33d2947e8ead304969849e64f2a6    --staticnode:/ip4/127.0.0.1/tcp/60001/p2p/16Uiu2HAkyTos6LeGrj1YJyA3WYzp9qKQGCsxbtvyoBRHSu9PCrQZ --ports-shift=9 --dns-discovery=false --rln-relay-eth-account-address:xxx --rln-relay-eth-client-address:xxx    --rln-relay-eth-account-private-key:xxx
```

### nwaku version/commit hash
`v0.12.0-2-g8a8c74`

## Acceptance criteria
- [ ] Add feature allowing verification against provided roots

**Comment by rymnc** [comment]
> I don't understand why this is considered a bug and why we don't want anymore this checks. Zerokit is in fact relying on merkle tree state by design: more precisely it does the following checks on proof values https://github.com/vacp2p/zerokit/blob/5d429ca0311752f6c2052a029cefad32d0ae1f7a/rln/src/public.rs#L327-L331 . Removing the root check seems a regression to me. Nodes and users should listen to the same contract, otherwise how we can ensure that users paid correctly the membership fee?
> 

You're right. Should we add the window of acceptable roots to zerokit then?

**Comment by rymnc** [comment]
Fixed by #1256 and #1250
